File: week2_algorithmic_warmup/3_greatest_common_divisor.js
const greatest_common_divisor_fast = function(a,b) {
  let aa = Math.max(a, b)
  let bb = Math.min(a, b)
  let divisor = aa % bb
  if(bb === 0) return aa
  return greatest_common_divisor_fast(bb,divisor)
}

data = '';

process.stdin.on('readable', function() {
  var chunk;
  while (chunk = process.stdin.read()) {
    data += chunk;
  }
});

process.stdin.on('end', function () {

  let lines = data.split(/\r\n|\r|\n|\\n/)
  let input = lines[0].split(' ')

  console.log(greatest_common_divisor_fast(input[0], input[1]))
});

File: week2_algorithmic_warmup/3_greatest_common_divisor.py
# Uses python3
import sys

def greatest_common_divisor_fast(a, b):
    aa = max(a, b)
    bb = min(a, b)
    if bb == 0:
        return aa
    divisor = aa % bb
    return greatest_common_divisor_fast(bb, divisor)

if __name__ == '__main__':
    input = sys.stdin.read();
    a, b = map(int, input.split())
    print(greatest_common_divisor_fast(a, b))
